### PR TITLE
[Enhancement] Add option to always get the Cremia Hug Cutscene after completing the milk run

### PR DESCRIPTION
**Pull request by balloondude2**
After obtaining the Romani Mask, completing the Milk Run minigame will randomly give you either a nice cutscene or 200 rupees. This PR will add an enhancement that will let you choose what reward you get. 

### Build Artifacts
  - [2ship-linux.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2027478330.zip)
  - [2ship-windows.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2027481394.zip)
  - [2ship-mac.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/2027507751.zip)

### CODE REVIEW

**Comment by SuperBro64** [comment]
It might be an interesting idea if for more flexibility this was instead a dropdown where you could choose to guarantee either the hug or the 200 Rupees. The dropdown could be called something like "Cremia's Reward After Romani's Mask", with the available options being:

- "Random" (default option)
- "Always Hug"
- "Always Huge Rupee"

Just a suggestion.

**Comment by balloondude2** [comment]
That's a good idea. I've updated the enhancement to a list of options. 

I haven't tested whether the new options work as intended yet though. 

If anyone wants to test it out, you can set weekevent 52x00 after starting the mini game and before getting to milk road to skip the section with the Gorman Brothers.

**Comment by balloondude2** [comment]
Confirmed that you are able to get both rewards with the vanilla option, and get the appropriate reward with the other options.

File: mm/src/overlays/actors/ovl_En_Ma_Yto/z_en_ma_yto.c [hunk 1]
```diff
@@ -308,7 +310,8 @@ void EnMaYto_ChooseAction(EnMaYto* this, PlayState* play) {
         case MA_YTO_TYPE_AFTERMILKRUN:
             this->unk310 = 0;
             if ((INV_CONTENT(ITEM_MASK_ROMANI) == ITEM_MASK_ROMANI) &&
-                CHECK_WEEKEVENTREG(WEEKEVENTREG_ESCORTED_CREMIA) && (Rand_Next() & 0x80)) {
+                CHECK_WEEKEVENTREG(WEEKEVENTREG_ESCORTED_CREMIA) &&
+                (GameInteractor_Should(GI_VB_RANDOM_COMPARISON, Rand_Next() & 0x80, NULL))) {
```
Comment (garrettjoecox) [inline mm/src/overlays/actors/ovl_En_Ma_Yto/z_en_ma_yto.c]:
```suggestion
                (GameInteractor_Should(GI_VB_RANDOM_COMPARISON, Rand_Next() & 0x80))) {
```

File: mm/2s2h/Enhancements/Cutscenes/Cutscenes.cpp [hunk 2]
```diff
@@ -13,4 +13,6 @@ void RegisterCutscenes() {
     RegisterSkipEntranceCutscenes();
     RegisterSkipIntroSequence();
     RegisterSkipToFileSelect();
+
+    RegisterCremiaHugs();
```
Comment (garrettjoecox) [inline mm/2s2h/Enhancements/Cutscenes/Cutscenes.cpp]: Does this make more sense in the `Minigames` category? (does a minigame trigger this?)

File: mm/2s2h/Enhancements/GameInteractor/GameInteractor.h [hunk 3]
```diff
@@ -55,6 +55,7 @@ typedef enum {
     GI_VB_FLIP_HOP_VARIABLE,
     GI_VB_DISABLE_LETTERBOX,
     GI_VB_CLOCK_TOWER_OPENING_CONSIDER_THIS_FIRST_CYCLE,
+    GI_VB_RANDOM_COMPARISON,
```
Comment (garrettjoecox) [inline mm/2s2h/Enhancements/GameInteractor/GameInteractor.h]:
```suggestion
    GI_VB_PLAY_CREMIA_HUG_CUTSCENE,
```

File: mm/2s2h/Enhancements/Cutscenes/Cutscenes.cpp (hunk 2 above)
Comment (balloondude2) [reply mm/2s2h/Enhancements/Cutscenes/Cutscenes.cpp]: The minigame category makes more sense. I moved it there.

**Comment by balloondude2** [comment]
Just retested that the enhancement is still working after moving things around and adding an entry to the modern menu. I also updated the option's wording to be more concise by adding a tooltip. All of the comments should have been resolved.

**Review by garrettjoecox — APPROVED** [review]

**Pull request merged**